api/src/main/java/org/openmrs/module/fhir/api/strategies/relatedperson/RelatedPersonStrategy.java

```diff
@@ -27,6 +29,28 @@ public RelatedPerson getRelatedPerson(String uuid) {
 		return FHIRRelatedPersonUtil.generateRelationshipObject(omrsRelationship);
 	}
 
+	@Override
+	public List<RelatedPerson> searchByPatientIdentifier(String identifierValue) {
+		List<RelatedPerson> relatedPersonList = new ArrayList<>();
+		List<Relationship> relationships = new ArrayList<>();
+		List<PatientIdentifierType> allPatientIdentifierTypes = Context.getPatientService().getAllPatientIdentifierTypes();
+		List<org.openmrs.Patient> patientList = Context.getPatientService().getPatients(identifierValue,
+				null, allPatientIdentifierTypes, true);
+
+		if (patientList != null && !patientList.isEmpty()) {
+			for (Patient patient : patientList) {
+				relationships.addAll(Context.getPersonService().getRelationshipsByPerson(patient));
```

Comment:
Yeah, you are right.
I was relying on methods implemented in the openmrs core for getting relationships. There is none for getting by patient identifier. That's I did the above.

So, for me to achieve one DB request, I have to write a hibernate native query. That's what I will be working on. If you have a better way to achieve this, I would know

Reply:
Like I commented before - it might be okay to have `n` database requests since we expect usually one patient or at least not very many to come back since we're searching by identifier. Might be okay to proceed without the new hiberate query.